dbt/include/fabric/macros/materializations/models/table/table.sql

```diff
@@ -32,13 +25,24 @@
   -- `BEGIN` happens here:
   {{ run_hooks(pre_hooks, inside_transaction=True) }}
 
+  -- naming a temp relation
+  {% set tmp_relation = target_relation.incorporate(path={"identifier": target_relation.identifier ~ '__dbt_tmp_vw'}, type='view')-%}
```

Comment:
You could use `make_temp_relation` for this (though you'd still need to set the `type='view'` after)